docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Database
+
+The database reads in SEP module data from data.csv and stores the useful data into an ArrayList of module mappings and universities.
+
+Upon starting easySEP, the DatabaseStorage will load each line from data.csv, parse the line using DatabaseParser, and store the data into the Database.
+
+Relevant exceptions are thrown when there are unexpected scenarios. For instance, if data.csv cannot be found at the given file path, a FileNotFoundException is thrown.
+
+The following diagram illustrates the relationships between the three main database classes - DatabaseStorage, DatabaseParser, and Database.
+
+![Database Class Diagram](./images/Database_Class.png)
```

Comment:
Consider changing from square bullets to '+' and '-' to show public and private methods
![image](https://user-images.githubusercontent.com/88868259/197936164-68b2e9b9-93f0-4d68-a0ad-6dff011522b1.png)